### ISSUE TITLE
Bug: Segfault state_history_plugin when fetch_block: true

### ISSUE THREAD

**Issue by vtsurka** [EOSIO/eos#7136]
I run node from snapshot and exists state-history. If in get_blocks_request_v0 we set fetch_block: true it will be throw segfault. 
Code from [state_history_plugin.cpp](https://github.com/EOSIO/eos/blob/master/plugins/state_history_plugin/state_history_plugin.cpp#L82)
```C++
   void get_block(uint32_t block_num, fc::optional<bytes>& result) {
      chain::signed_block_ptr p;
      try {
         p = chain_plug->chain().fetch_block_by_number(block_num);
      } catch (...) {
         return;
      }
      result = fc::raw::pack(*p);
   }
```

I think we need to add condition on null result and maybe some warning log
```C++
if (p) {
      result = fc::raw::pack(*p);
}
```

**Comment by taokayan** [comment]
nice catch @vtsurka

**Comment by heifner** [comment]
Fixed by #7384
